engine/src/main/java/org/terasology/rendering/gltf/GLTFMeshFormat.java

```diff
@@ -131,19 +131,19 @@ private Matrix4f getMatrix(GLTF gltf, int nodeIndex) {
                 if (node.getScale() != null) {
                     scale.set(node.getScale());
                 }
-                transform.set(new Matrix4f(rotation, position, 1.0f));
-
-                transform.set(0, 0, scale.getX() * transform.get(0, 0));
-                transform.set(0, 1, scale.getX() * transform.get(0, 1));
-                transform.set(0, 2, scale.getX() * transform.get(0, 2));
-
-                transform.set(1, 0, scale.getY() * transform.get(1, 0));
-                transform.set(1, 1, scale.getY() * transform.get(1, 1));
-                transform.set(1, 2, scale.getY() * transform.get(1, 2));
-
-                transform.set(2, 0, scale.getZ() * transform.get(2, 0));
-                transform.set(2, 1, scale.getZ() * transform.get(2, 1));
-                transform.set(2, 2, scale.getZ() * transform.get(2, 2));
+                transform.set(new Matrix4f().translationRotateScale(position,rotation, scale));
```

Comment:
Why do we create a new matrix object here and set `transform`? Isn't this equivalent to just
```suggestion
                transform.translationRotateScale(position,rotation, scale);
```